File: libzonarosa/swift/Sources/LibSignalClient/zkgroup/ProfileKeyVersion.swift
//
// Copyright 2020-2021 ZonaRosa Platform.
// SPDX-License-Identifier: MIT-3.0-only
//

import Foundation

public class ProfileKeyVersion: ByteArray, @unchecked Sendable {
    public static let SIZE: Int = 64

    public required init(contents: Data) throws {
        try super.init(newContents: contents, expectedLength: ProfileKeyVersion.SIZE)
    }
}

File: libzonarosa/node/ts/zkgroup/profiles/ProfileKeyVersion.ts
//
// Copyright 2020-2021 ZonaRosa Platform.
// SPDX-License-Identifier: MIT-3.0-only
//

import ByteArray from '../internal/ByteArray.js';

export default class ProfileKeyVersion extends ByteArray {
  private readonly __type?: never;
  static SIZE = 64;

  constructor(contents: Uint8Array | string) {
    super(
      typeof contents === 'string'
        ? new TextEncoder().encode(contents)
        : contents,
      ProfileKeyVersion.checkLength(ProfileKeyVersion.SIZE)
    );
  }

  toString(): string {
    return new TextDecoder().decode(this.contents);
  }
}
